eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelEvents.scala

```diff
@@ -36,7 +36,7 @@ case class ChannelRestored(channel: ActorRef, peer: ActorRef, remoteNodeId: Publ
 
 case class ChannelIdAssigned(channel: ActorRef, remoteNodeId: PublicKey, temporaryChannelId: ByteVector32, channelId: ByteVector32) extends ChannelEvent
 
-case class ShortChannelIdAssigned(channel: ActorRef, channelId: ByteVector32, shortChannelId: ShortChannelId) extends ChannelEvent
+case class ShortChannelIdAssigned(channel: ActorRef, channelId: ByteVector32, shortChannelId: ShortChannelId, previousShortChannelId: Option[ShortChannelId] = None) extends ChannelEvent
```

Comment:
Why is `previousShortChannelId` optional if it can always be defined?

Reply:
It's only defined if you had a *different* previous shortChannelId.
Do you mean that we should instead always provide it, and have the same value in `shortChannelId` and `previousShortChannelId` when it doesn't change?
It feels a bit wasteful to me because it happens very infrequently, and would require changing a bit more code.

Reply:
No I just missed the fact that you had defined a default `None` value. I tend to avoid that as much as possible because it is easier to estimate impacts on existing code.

Reply:
Got it, I can remove that default `None` and explicitly set it when this event is created, would that be ok?